metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java

```diff
@@ -1731,4 +1795,75 @@ public T getRequiredObjectProperty(T object, String name) {
      */
     public abstract void putSort(T object,String name,Sort s);
 
+
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+    // The above abstract methods were added based on need, and they sufffer badly from a strict top-down approach.
+    // There are multiple methods that do the same thing, and there are missing methods for crucial functionaloty.
+    // Don't use them. Use the ones below. 
+    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////    
```

Comment:
I deprecated them. The parser still uses them,and there are other
extensions that still rely on these. Added
https://github.com/lightblue-platform/lightblue-core/issues/625

On Thu, Mar 31, 2016 at 5:49 PM, Dennis Crissman <notifications@github.com>
wrote:

> In
> metadata/src/main/java/com/redhat/lightblue/metadata/parser/MetadataParser.java
> <https://github.com/lightblue-platform/lightblue-core/pull/624#discussion_r58145871>
> :
>
> > @@ -1731,4 +1795,75 @@ public T getRequiredObjectProperty(T object, String name) {
> >       */
> >      public abstract void putSort(T object,String name,Sort s);
> >
> > +
> > +    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
> > +    // The above abstract methods were added based on need, and they sufffer badly from a strict top-down approach.
> > +    // There are multiple methods that do the same thing, and there are missing methods for crucial functionaloty.
> > +    // Don't use them. Use the ones below.
> > +    ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
>
> Should the above abstract methods be deprecated or have javadoc noting
> that they are not preferred? It might be useful to note what method is
> preferred for each.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/lightblue-platform/lightblue-core/pull/624/files/ea3a82c505fb57a14479315fd5b5ae19d4fdd0db#r58145871>
>